workflow-templates/im-deploy-az-app-manually.yml

```diff
@@ -33,8 +33,8 @@ on:
       tag:
         description: The tag for the release that will be deployed.  For Production, only tags reachable by the default branch will be accepted.
         required: true
-      environment:
-        description: The environment to deploy to
+      environment-or-target:
```

Comment:
I can tolerate this name.

Reply:
I'm not sure how I feel about the environment-or-target name.  The description says the environment or target to deploy to.  That kind of makes it seem like it's either or, but it's just a subset of the environment.  I haven't looked at it myself but would providing a second input that targets the region (primary/secondary) make this clearer rather than modifying the environment?

Reply:
An additional input to select the region isn't quite right either, because only Stage and Prod environments have both regions as an option. Dev, QA, Stage, and Demo only have the primary region.

This is why I favored `destination`. That way the variable _name_ doesn't imply anything about whether it's a GitHub environment from which to get secrets, whether it's a primary or secondary or any other aspect. It only implies that it's a known destination, whether that destination is a whole single-region environment, whether its the primary region of an environment, the secondary region of an environment, or even a customized workflow might deploy to multiple regions of an environment simultaneously. It's just as reasonable, in my mind, to say "the deploy destination is 'dev'" as it is to say "the deploy destination is 'stage-secondary'".

Another possible name for this input could just be `deploy-to`.

Reply:
I have updated the description.  I feel if we aren't using environment, then there should not be any variables referencing environment in all the repos. Why do we use environment at all.  Not having environment will not be intuitive to all users.  I conceded to add target, but teams will call it an environment regardless.  They are used to the environment terminology.  I think there needs to be some give here instead being 100% syntactically correct with this.